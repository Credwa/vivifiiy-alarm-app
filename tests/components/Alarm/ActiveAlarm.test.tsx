import React from 'react';
import renderer from 'react-test-renderer';
import ActiveAlarms from '@/components/Alarm/ActiveAlarm';
import { AlarmInterface } from '@/interfaces';

describe('<ActiveAlarms />', () => {
  it(`renders correctly`, () => {
    const mockActiveAlarm: AlarmInterface = { active: true, key: '0730AM', hour: '07', minute: '30', meridiem: 'AM' };
    const tree = renderer.create(<ActiveAlarms nearestAlarm={mockActiveAlarm} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
