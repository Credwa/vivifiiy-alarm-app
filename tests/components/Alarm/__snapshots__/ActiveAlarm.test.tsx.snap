// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveAlarms /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#869DB2",
        },
        Object {
          "fontFamily": "Nunito_400Regular",
          "fontSize": 28,
        },
        Object {
          "color": "#B1C2D1",
        },
        undefined,
      ]
    }
  >
    Next alarm at...
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 30,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#869DB2",
          },
          Object {
            "fontFamily": "Nunito_400Regular",
            "fontSize": 48,
          },
          Object {
            "color": "#CEE1F2",
          },
          undefined,
        ]
      }
    >
      07
       : 
      30
       
      <Text
        style={
          Array [
            Object {
              "color": "#869DB2",
            },
            Object {
              "fontFamily": "Nunito_400Regular",
              "fontSize": 44,
            },
            Object {
              "color": "#CEE1F2",
            },
            undefined,
          ]
        }
      >
        AM
      </Text>
    </Text>
  </View>
</View>
`;
